app/models/concerns/roleable.rb

```diff
@@ -18,6 +18,10 @@ def regular?
     !staff?
   end
 
+  def whisperer?
+    SiteSetting.enable_whispers.present? && groups&.exists?(id: SiteSetting.enable_whispers.split("|"))
```

Comment:
```suggestion
    SiteSetting.enable_whispers.present? && group_users&.exists?(id: SiteSetting.enable_whispers.split("|"))
```

This avoids an extra joins to the `groups` table.

Reply:
I think we should also memoize this to avoid the extra SQL query each time this is called on the same user instance.